src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private static final double INCREASE_ARRAY = 1.5;
+    private Object[] arraylist = new Object[ARRAY_SIZE];
+    private int size;
+
+    public ArrayList() {
+        this.arraylist = arraylist;
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 <= arraylist.length) {
+            arraylist[size] = value;
+        } else if (size + 1 > ARRAY_SIZE) {
+            grow();
+            arraylist[size] = value;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        if (size + 1 == arraylist.length) {
+            grow();
+        }
+        System.arraycopy(arraylist,index,arraylist,index + 1,size - index);
+        arraylist[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        Object [] newArray = new Object[ (int)(arraylist.length * INCREASE_ARRAY)];
```

Comment:
fix whitespaces - remove redundant ones and add whitespace after cast to int

Reply:
Done.